### PR TITLE
[OOP][V4] Update PS and Java bookworm images to 4.1035.0

### LINKED ISSUE #1102: Update PowerShell image to use 10XX host version
When were ready to roll it out, we will need to update the PowerShell Dockerfiles' `HOST_VERSION` arguments to use the 4.10XX version of the host. This change will apply to all Linux SKUs.

### LINKED ISSUE #1103: Update Java image to use 10XX host version
When were ready to roll it out, we will need to update the Java Dockerfiles' `HOST_VERSION` arguments to use the 4.10XX version of the host. This change will apply to all Linux SKUs.

### PR DESCRIPTION
**Pull request by pragnagopa**
Fixes #1102 Fixes #1103

### CODE REVIEW

File: host/4/bookworm/java/java21/java21-appservice.Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ARG JAVA_VERSION=21.0.1
 ARG JAVA_HOME=/usr/lib/jvm/msft-21-x64
 # Build the runtime from source
-ARG HOST_VERSION=4.34.2
+ARG HOST_VERSION=4.1034.2
```
Comment (kshyju) [inline host/4/bookworm/java/java21/java21-appservice.Dockerfile]: FYI - I will create the new tag/release EOD. 4.1035.0

Comment (pragnagopa) [reply host/4/bookworm/java/java21/java21-appservice.Dockerfile]: done updated

File: host/4/bookworm/java/java21/java21-appservice.Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ARG JAVA_VERSION=21.0.1
 ARG JAVA_HOME=/usr/lib/jvm/msft-21-x64
 # Build the runtime from source
-ARG HOST_VERSION=4.34.2
+ARG HOST_VERSION=4.1035.0
```
Comment (kshyju) [inline host/4/bookworm/java/java21/java21-appservice.Dockerfile]: The final stage (line 43) still appears to be using .NET 6. Shouldn't it be using .NET 8.0 instead?
oop-host [TFM is net8.0](https://github.com/Azure/azure-functions-host/blob/634627a86b38e1dd10dbf0748e7331e707f9da87/src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj#L5)

File: host/4/bookworm/java/java21/java21-appservice.Dockerfile [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
 FROM mcr.microsoft.com/dotnet/sdk:6.0-bookworm-slim-amd64 AS runtime-image
 ARG HOST_VERSION
 COPY --from=mcr.microsoft.com/dotnet/sdk:8.0-bookworm-slim-amd64 [ "/usr/share/dotnet", "/usr/share/dotnet" ]
```
Comment (amamounelsayed) [inline host/4/bookworm/java/java21/java21-appservice.Dockerfile:7]: Is there a reason to still depend on FROM mcr.microsoft.com/dotnet/sdk:6.0-bookworm-slim-amd64 AS runtime-image?

File: host/4/bookworm/java/java21/java21-appservice.Dockerfile [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ARG JAVA_VERSION=21.0.1
 ARG JAVA_HOME=/usr/lib/jvm/msft-21-x64
 # Build the runtime from source
-ARG HOST_VERSION=4.34.2
+ARG HOST_VERSION=4.1035.0
 FROM mcr.microsoft.com/dotnet/sdk:6.0-bookworm-slim-amd64 AS runtime-image
```
Comment (amamounelsayed) [inline host/4/bookworm/java/java21/java21-appservice.Dockerfile:5]: Is there a reason to still depend on FROM mcr.microsoft.com/dotnet/sdk:6.0-bookworm-slim-amd64 AS runtime-image?

File: host/4/bookworm/java/java21/java21-appservice.Dockerfile (hunk 2 above)
Comment (kshyju) [reply host/4/bookworm/java/java21/java21-appservice.Dockerfile]: ~~Could you please use 4.1035.1 ? ~~ I pushed an update with the new host version.

**Comment by kshyju** [comment]
I pushed an iteration to update the host version to 4.1035.1

**Comment by pragnagopa** [comment]
Replaced by https://github.com/Azure/azure-functions-docker/pull/1118

**Pull request closed without merging**